trunk/branches/tags not always the first entry in the relative URL

cdonnellytx opened the pull request:
Fixed problem where it assumed `trunk`/`branches`/`tags` was always the first entry in the relative URL.  That is not always true - a repo can have multiple directories each with their own `trunk`/`branches`/`tags`.

Commit:
Fixed problem where it assumed trunk/branches/tags had to be the first entry in the relative URL.  That is not always true.

Commit:
add functionality to enable including external file statuses, disable remote

Commit:
Reworked svn directory detection to not call "svn info", as that added
to the global error count, which may or may not be tracked by user
profiles.

Commit:
Do not dot source, it is slow

Commit:
more "do not dot source"

Commit:
Several improvements

- Refactored OnRemove to remove any scripts with Write-SvnStatus.
- Now writes full URL when trunk/branches/tags not found.

Commit:
updated title, branch default is now relative URL

Commit:
fixed some broken things

Commit:
use root relative URL for repo name

Commit:
formatting.  Note at this point I am considering myself forked from the parent

Commit:
revamp to make better

Commit:
201: fix the colors again.

Commit:
Fixed detection of modified properties

Commit:
fixed unindexed labels

Commit:
posh-svn now wraps svn binary so we can paginate svn help/log, if at all possible

Commit:
oops forgot to remove debug

Commit:
also do diff, always come up even if less than 1 screen (the way git does)

Commit:
fixed Get-SvnDirectory so it stops looking for .svn files, does not return .svn directory

Commit:
making whitespace consistent

Commit:
Removed dependency on personal module.

This no longer depends on my personal module "ConsoleTheme".
Eventually I plan on releasing it, although it needs a shakedown before I do that
(e.g., is "ConsoleTheme" even the right name, when it's not prepackaged themes, but more utilities?)

Commit:
fixing bug where PowerShell 2.0 fails because ScriptStackTrace does not exist in that version.

Commit:
wow PowerShell 2.0 does not support anything

Commit:
ensure PAGER does not show BOMs

cdonnellytx committed:
Stripping out my ConsoleTheme entirely.

withinfocus commented on posh-svn.psm1:
```diff
@@ -1,15 +1,87 @@
+    }
+}
+
+# TODO when was ScriptStackTrace added? It's not in 2.0...
```
Planning on fixing?

cdonnellytx replied:
Yeah, I just need to double-check when this was added and add the minimum version check to guard it.  It's optional, so having it on PowerShell 2.0 isn't necessary.

withinfocus commented on posh-svn.psm1:
```diff
@@ -1,15 +1,87 @@
+    }
+
+    # HACK determine a minimum required version, 1.6.0 is a guess
+    $requiredVersion = [Version]'1.6.0'
```
Do you want to bump this up? 1.6 is ancient.

cdonnellytx replied:
I don't know what a minimum "reasonable" version is, I figured the syntax worked on 1.6 though.  We can bump higher if you'd rather.

withinfocus commented on posh-svn.psd1:
```diff
@@ -0,0 +1,56 @@
+ModuleToProcess = 'posh-svn.psm1'
+
+# Version number of this module.
+ModuleVersion = '0.7.301'
```
Where does this originate from?

cdonnellytx replied:
Sorry... I put in 0.7 to match posh-git on my local and the higher number was to avoid it being overwritten if I updated the module for whatever reason.  I'll change it to whatever makes the most sense.

withinfocus commented on posh-svn.psd1:
```diff
@@ -0,0 +1,56 @@
+GUID = 'f18820b6-2e02-41b4-afc5-de886bb1b848'
+
+# Author of this module
+Author = 'Chris Donnelly, Jeremy Skinner and contributors'
```
We'd appreciate being included as the primary author, but fine with the rest.

cdonnellytx replied:
Yup.  Sorry - I was forking at one point instead of merging back in (hence all my changes to include my personal modules).  This was one of the changes.

I feel bad about this one, really shouldn't have put my name first.

cdonnellytx replied:
Could also just leave my name off too and be one of the contributors.

withinfocus commented on posh-svn.psd1:
```diff
@@ -0,0 +1,56 @@
+Author = 'Chris Donnelly, Jeremy Skinner and contributors'
+
+# Copyright statement for this module
+Copyright = '(c) 2010-2017 Chris Donnelly, Jeremy Skinner and contributors'
```
We don't have a copyright in place and would actually prefer that it be eliminated for this project.

cdonnellytx replied:
Sure.  Just force of habit, will remove.

withinfocus reviewed (changes requested):
Great work, thanks. A few comments and questions.

cdonnellytx committed:
Cleanup (#1)

Cleaning up problems with PR.

- Removed copyright
- Fixed names to be the owners of the repo (will need to double-check)
- Version number now less wonky
- Realized I had duplicated CheckVersion.ps1's code in the script, moved the logic to the proper script
- Fixed PS 2.0 script loading (it needs dot sourcing)
- Fixed error handling so 2.0/3.0+ differences are now isolated.

withinfocus reviewed (approved):
I think "Various contributors" alone would be fine for the author, not a worry of ours really. Looks good though.

cdonnellytx committed:
Ignoring error
